### ISSUE TITLE
Major increase in power consumption

### ISSUE THREAD

**Issue by rymc** [siacs/Conversations#1359]
For Conversations (1.6.3) I have noticed increased power consumption, so much so that it now the number one user of power on my device. Previous versions used very little.

From ADB I can see that the following is output constantly:
```
AxolotlService (user@server): Received PEP device list update from user@server, processing...
AxolotlService (user@server): Received PEP device list update from user@server, processing...
AxolotlService (user@server): Received PEP device list update from user@server, processing...
AxolotlService (user@server): Own device <removednumber> not in PEP devicelist. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.devicelist"><item><list xmlns="eu.siacs.conversations.axolotl"><device id="<removednumber>"/></list></item></publish></pubsub></iq>
AxolotlService (user@server): Own device <removednumber> not in PEP devicelist. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.devicelist"><item><list xmlns="eu.siacs.conversations.axolotl"><device id="<removednumber>"/></list></item></publish></pubsub></iq>
AxolotlService (user@server): Own device <removednumber> not in PEP devicelist. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.devicelist"><item><list xmlns="eu.siacs.conversations.axolotl"><device id="<removednumber>"/></list></item></publish></pubsub></iq>

```

Disabling the account causes the log messages above to stop. Re-enabling the account causes the log messages above to resume.

Conversations: 1.6.3
Android: 5.1.1
Ejabberd: 2.1.10-4+deb7u1

**Comment by iNPUTmice** [comment]
possibly related to #1358 and old ejabberds doing weird stuff with PEP.

**Comment by gnuheidix** [comment]
Two of my relatives are also using Conversations 1.6.3 and are also complaining about this issue.

**Comment by rymc** [comment]
I'm not sure if it's related to #1358 as I am not seeing any increase in the size of the ejabberd logs.

**Comment by iNPUTmice** [comment]
@rymc the increased logs might just be a symptom. That user has probably enabled debug logs and you do not.

@gnuheidix what server are you using?

**Comment by gnuheidix** [comment]
I'm using Ubuntu 14.04 latest - ejabberd 2.1.11.

**Comment by iNPUTmice** [comment]
We will try to come up with a work around but you will be way better of if you would update your ejabberd server.

**Comment by iNPUTmice** [comment]
Also if any of you could provide me with a account on your server that would be appreciated. Mail to: inputmice@siacs.eu

**Comment by rymc** [comment]
I've just sent you an email with account details.

**Comment by gnuheidix** [comment]
I've also sent you an email regarding your account request. :-)

**Comment by iNPUTmice** [comment]
Can you update to 1.7.0-beta (available on the  beta channel or in the development branch). I'm not able to reproduce the logout above.

**Comment by rymc** [comment]
I just built and installed 1.7.0-beta and I can reproduce the output above.

**Comment by iNPUTmice** [comment]
are you using multiple devices?

**Comment by rymc** [comment]
I had a client running on my computer (mcabber) but I just tried turning of mcabber, then disabling and re-enabling my conversations application.  I also tried deleting the account from conversations and re-adding it. Problem still persists.

**Comment by iNPUTmice** [comment]
Ok. Thanks for the accounts.

This way worse than I originally thought. @strb just pushed some changes b84fecf51abd0f6e02d49307827622b5aa6f4044 that might ease the symptoms a little bit but there is just no way in recovering once an account has been broken.

Prepare yourself to update your ejabberd servers (You can use the [APT repository from jabber.at]( https://jabber.at/en/apt-repository)) or at the very least disable PEP in your current server. If this is not your server pick a different one.

If you have bought the App in the PlayStore feel free to send me a refund request.

**Comment by rymc** [comment]
I've just disabled PEP on the server and the issue is gone. That's good enough for me. 

Thanks.

**Comment by danielreuther** [comment]
Same problem on OpenFire, apparently. Android is telling me Conversations used 122MB of bandwidth over the course of the last two days. My phone's battery is dying in a couple of hours.

@iNPUTmice Do you know what change is causing this to be an issue in 1.6?
I'll probably not be able to migrate to another server any time soon.

Did anyone test this on Tigase, yet? Are you seeing the same symptoms?

**Comment by iNPUTmice** [comment]
does the openfire problem remain with 1.6.3?

the ejabberd problem was being caused by a faulty pep implementation. I'm
not aware of any bugs in openfire that could cause a similar problem.

however an adb logcat output will give us more details.

**Comment by danielreuther** [comment]
I currently have 1.6.3 installed, yes. This is on OF 3.9.3.

It's really bad, though. Conversations is not only killing my battery but also my bandwidth and makes my phone behave sluggish.

Running an adb logcat results in the following message sequence being repeated every 200-400ms:
```
08-25 17:37:13.068 D/conversations( 2725): AxolotlService (user@example.com): Received PEP device list update from user@example.com, processing...
08-25 17:37:13.201 D/conversations( 2725): AxolotlService (user@example.com): Received PEP device list update from user@example.com, processing...
08-25 17:37:13.286 D/conversations( 2725): AxolotlService (user@example.com): Own device 475996953 not in PEP devicelist. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.devicelist"><item><list xmlns="eu.siacs.conversations.axolotl"><device id="475996953"/></list></item></publish></pubsub></iq>
08-25 17:37:13.365 D/conversations( 2725): AxolotlService (user@example.com): Own device 475996953 not in PEP devicelist. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.devicelist"><item><list xmlns="eu.siacs.conversations.axolotl"><device id="475996953"/></list></item></publish></pubsub></iq>
```

**Comment by MiskyWhixer** [comment]
Same problem here with Conversations 1.6.3 und OpenFire Server 3.10.2 .

OpenFire debug.log : 

```
2015.08.25 17:46:33 org.jivesoftware.openfire.spi.RoutingTableImpl - Unable to route packet. No session is available so store offline. <message to="user1@beispiel.de" from="user2@beispiel.de" id="user2@beispiel.de__user1@beispiel.de__FMB1d"><event xmlns="http://jabber.org/protocol/pubsub#event"><items node="eu.siacs.conversations.axolotl.devicelist"><item id="3bc0a8a3-0417-435d-89e2-09e1d2c56a8828821" node="eu.siacs.conversations.axolotl.devicelist"><list xmlns="eu.siacs.conversations.axolotl"><device id="2028788733"/></list></item></items></event></message> 
2015.08.25 17:46:33 org.jivesoftware.openfire.spi.RoutingTableImpl - Failed to route packet to JID: user1@beispiel.de packet: <message to="user1@beispiel.de" from="user2@beispiel.de" id="user2@beispiel.de__user1@beispiel.de__FMB1d"><event xmlns="http://jabber.org/protocol/pubsub#event"><items node="eu.siacs.conversations.axolotl.devicelist"><item id="3bc0a8a3-0417-435d-89e2-09e1d2c56a8828821" node="eu.siacs.conversations.axolotl.devicelist"><list xmlns="eu.siacs.conversations.axolotl"><device id="2028788733"/></list></item></items></event></message>
```
...

**Comment by iNPUTmice** [comment]
@danielreuther @MiskyWhixer please try with latest development HEAD

**Comment by iNPUTmice** [comment]
1.6.5 is out. please report if this fixes your problems

**Comment by mhde** [comment]
1.6.5 fixed the issue for me (ejabberd 15.07).

**Comment by gnuheidix** [comment]
ejabberd 2.1.11: I've measured four megabytes of idle traffic in 30 Minutes even with Conversations 1.6.5. :-(

After disabling mod_pubsub in ejabberd, the problem seems fixed for my four Conversation clients.

**Comment by iNPUTmice** [comment]
@gnuheidix Yes for ejabberd this is a **WONTFIX**. If you use ejabberd < 14.12 you will have to disable pep / pubsub.

1.6.5 was just trying to fix a similar issue with openfire

**Comment by MiskyWhixer** [comment]
1.6.5 also fixed this issue with OpenFire 3.10.2
Thank you!

**Comment by danielreuther** [comment]
I'm not sure whether this is related to the previous PEP issues at all (highly doubt it, actually) but before I open you new issue I wanted to check back with you, @iNPUTmice.

The power consumption seems to have stabilized with 1.6.5 but increased significantly again with the latest version. So much so that my battery is warming up from it (Conversations is reported to be responsible for 75% of the drain).

I haven't changed any server-side configurations lately; which is why I presume that it must be a recent change in Conversations that's causing this. From the logcat output it looks like it periodically losing connection to the server. That's the output that is repeated every ~90secs:

```
09-17 09:47:23.632 D/conversations( 6545): user@exmaple.com: connecting
09-17 09:47:23.652 D/conversations( 6545): using dns server: ***
09-17 09:47:25.304 D/conversations( 6545): user@exmaple.com: using values from dns ***
09-17 09:47:29.398 D/conversations( 6545): user@exmaple.com: TLS connection established
09-17 09:47:29.519 D/conversations( 6545): user@exmaple.com/#resource: Authenticating with PLAIN
09-17 09:47:30.069 D/conversations( 6545): user@exmaple.com: logged in
09-17 09:47:30.249 D/conversations( 6545): user@exmaple.com: clearing 1 iq callbacks
09-17 09:47:30.249 D/conversations( 6545): AxolotlService (user@exmaple.com): Error received while publishing bundle: null
09-17 09:47:30.249 D/conversations( 6545): user@exmaple.com: done clearing iq callbacks. 0 left
09-17 09:47:30.490 D/conversations( 6545): user@exmaple.com: online with resource #resource
09-17 09:47:30.500 D/conversations( 6545): user@exmaple.com: fetching roster
09-17 09:47:32.392 D/conversations( 6545): AxolotlService : Couldn't find <item> in bundle IQ packet: <iq to="user@exmaple.com/#resource" id="*" xmlns="jabber:client" from="user@exmaple.com" type="error"><pubsub xmlns="http://jabber.org/protocol/pubsub"><items node="eu.siacs.conversations.axolotl.bundles:557376941" xmlns="http://jabber.org/protocol/pubsub"/></pubsub><error type="cancel" xmlns="jabber:client" code="503"><service-unavailable xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/></error></iq>
09-17 09:47:32.392 W/conversations( 6545): AxolotlService (user@exmaple.com): Received invalid bundle:<iq to="user@exmaple.com/#resource" id="*" xmlns="jabber:client" from="user@exmaple.com" type="error"><pubsub xmlns="http://jabber.org/protocol/pubsub"><items node="eu.siacs.conversations.axolotl.bundles:557376941" xmlns="http://jabber.org/protocol/pubsub"/></pubsub><error type="cancel" xmlns="jabber:client" code="503"><service-unavailable xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/></error></iq>
09-17 09:47:32.392 W/conversations( 6545): AxolotlService (user@exmaple.com): Received invalid prekeys:<iq to="user@exmaple.com/#resource" id="*" xmlns="jabber:client" from="user@exmaple.com" type="error"><pubsub xmlns="http://jabber.org/protocol/pubsub"><items node="eu.siacs.conversations.axolotl.bundles:557376941" xmlns="http://jabber.org/protocol/pubsub"/></pubsub><error type="cancel" xmlns="jabber:client" code="503"><service-unavailable xmlns="urn:ietf:params:xml:ns:xmpp-stanzas"/></error></iq>
09-17 09:47:32.392 I/conversations( 6545): AxolotlService (user@exmaple.com): Adding own IdentityKey org.whispersystems.libaxolotl.IdentityKey to PEP.
09-17 09:47:33.363 I/conversations( 6545): AxolotlService (user@exmaple.com): Adding new signedPreKey with ID #signedPreKeyId to PEP.
09-17 09:48:40.290 I/conversations( 6545): AxolotlService (user@exmaple.com): Adding 100 new preKeys to PEP.
09-17 09:48:40.300 D/conversations( 6545): AxolotlService (user@exmaple.com): : Bundle 557376941 in PEP not current. Publishing: <iq type="set"><pubsub xmlns="http://jabber.org/protocol/pubsub"><publish node="eu.siacs.conversations.axolotl.bundles:557376941"><item><bundle xmlns="eu.siacs.conversations.axolotl"><signedPreKeyPublic signedPreKeyId="#signedPreKeyId">#signedPreKeyPublic
09-17 09:48:40.300 D/conversations( 6545): </signedPreKeyPublic><signedPreKeySignature>#signedPreKeySignature
09-17 09:48:40.300 D/conversations( 6545): </signedPreKeySignature><identityKey>#identityKey
09-17 09:48:40.300 D/conversations( 6545): </identityKey><prekeys xmlns="eu.siacs.conversations.axolotl"><preKeyPublic preKeyId="#preKeyId">#preKeyPublic
09-17 09:48:40.300 D/conversations( 6545): </preKeyPublic><preKeyPublic preKeyId="#preKeyId">#preKeyPublic
09-17 09:48:40.300 D/conversations( 6545): </preKeyPublic><preKeyPublic preKeyI
09-17 09:48:40.300 D/conversations( 6545): user@exmaple.com: successfully enabled carbons
09-17 09:48:40.480 D/conversations( 6545): user@exmaple.com: Socket is closed
```

I haven't seen any messages related to this in the server-side logs.

Grateful for any hints!

**Comment by iNPUTmice** [comment]
> 09-17 09:48:40.300 D/conversations( 6545): user@exmaple.com: successfully enabled carbons
09-17 09:48:40.480 D/conversations( 6545): user@exmaple.com: Socket is closed

this looks a lot like the server is dropping the connection (possibly because/after carbons are enabled)

**Comment by danielreuther** [comment]
Agree. But it's weird that this is only starting to happen now—and without any log messages on the server side. What I found strange too (but it might as well just be a logging problem) is the line before the carbons are enabled; it seems to end abruptly:

```
09-17 09:48:40.300 D/conversations( 6545): </preKeyPublic><preKeyPublic preKeyI
```

I'll increase the server-side log level and see whether it yields anything meaningful.

Btw, there's a 1min delay between the following 2 log messages. I'm guessing it's doing some costly calculations inbetween—if so, does it need to happen on every reconnect?

```
09-17 09:47:33.363 I/conversations( 6545): AxolotlService (user@exmaple.com): Adding new signedPreKey with ID #signedPreKeyId to PEP.
09-17 09:48:40.290 I/conversations( 6545): AxolotlService (user@exmaple.com): Adding 100 new preKeys to PEP.
```

**Comment by iNPUTmice** [comment]
@danielreuther does the 'Server Info' in Conversations report PEP as being available?

**Comment by danielreuther** [comment]
Yes, it says 'available'.

But here's the thing: For some reason (don't really know what yet, I just blindly toggled Wifi on and off and killed Conversations and restarted it), it was able to succesfully connect and stay connected. And in this case it was showing PEP as 'broken'. After restarting the app again, though, I ended up in the same reconnect loop again.

**Comment by iNPUTmice** [comment]
yes broken is what it is supposed to show (at least with openfire)

I think the problems might be unrelated. maybe it was your wifi dropping
connections or something?

**Comment by danielreuther** [comment]
Just so that we're on the same page: What I described above (that Conversations was able to connect and shows PEP as 'broken') just happened exactly one time—and never again afterwards. I don't think this is related to my connection. After all, this affects multiple clients on different networks.

Bear in mind, though, that this only affects recent Conversations versions. For instance, 1.5.2 connects just fine—but upgrading that very device to 1.6.9 will get you into a reconnect loop.

Yesterday night I was playing with different options on the server side to see whether this has any effect on the behavior of the app.
a) Disabling PEP on the server side seems to have no effect at all. I still see the same messages from Axolotl re PEP, App still shows it as 'available' and of course the connection is still dying.
b) A couple of times I got an error that Conversations wasn't able to parse the user's avatar right before the socket was closed. Just to test if it was related, I disabled VCards; unfortunately, without any effect.

So, no luck at all. Newer Conversations versions still cause a reconnect loop. Mid-term I'll need to look into migrating off of Openfire, I suppose. Just not sure when I'll be able to...
